src/core/Akka.Cluster/SBR/SplitBrainResolver.cs

```diff
@@ -25,23 +25,47 @@ namespace Akka.Cluster.SBR
     /// </summary>
     internal class SplitBrainResolver : SplitBrainResolverBase
     {
-        private readonly Cluster cluster;
+        private sealed class AcquireCluster : INoSerializationVerificationNeeded
+        {
+            public static readonly AcquireCluster Instance = new AcquireCluster();
+            private AcquireCluster(){}
+        }
+
+        private const string AcquireClusterKey = "acquire-cluster-key";
+        private Cluster _cluster;
 
         public SplitBrainResolver(TimeSpan stableAfter, DowningStrategy strategy)
             : base(stableAfter, strategy)
         {
-            cluster = Cluster.Get(Context.System);
-            Log.Info(
-                "SBR started. Config: strategy [{0}], stable-after [{1}], down-all-when-unstable [{2}], selfUniqueAddress [{3}].",
-                Logging.SimpleName(strategy.GetType()),
-                stableAfter,
-                // ReSharper disable VirtualMemberCallInConstructor
-                DownAllWhenUnstable == TimeSpan.Zero ? "off" : DownAllWhenUnstable.ToString(),
-                SelfUniqueAddress.Address);
+            
             // ReSharper restore VirtualMemberCallInConstructor
+            
+            Context.Become(WaitingForCluster);
```

Comment:
Might be better a better way of doing this other than behavior-switching in the constructor.